packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -366,6 +360,115 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentNestedTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, nestedStackName: string): Promise<Template> {
+    return this.readCurrentStackTemplate(stackArtifact, nestedStackName);
+  }
+
+  private async replaceNestedStacksInRootStack(rootStackArtifact: cxapi.CloudFormationStackArtifact, deployedTemplate: any) {
+    if (!this.stackHasNestedStacks(rootStackArtifact)) {
+      return;
+    }
+
+    const sdkWithRole = await prepareSdkWithLookupRoleFor(this.sdkProvider, rootStackArtifact);
+    const listStackResourcesByStackName: { [key:string] : ListStackResources } = {};
+    await this.replaceNestedStacksInParentTemplate(
+      rootStackArtifact, rootStackArtifact.template, deployedTemplate, rootStackArtifact.stackName, listStackResourcesByStackName, sdkWithRole.sdk,
+    );
+  }
+
+  private stackHasNestedStacks(stackArtifact: cxapi.CloudFormationStackArtifact): boolean {
+    return Object.values(stackArtifact.template.Resources ?? {}).some((resource: any) => resource.Type === 'AWS::CloudFormation::Stack');
+  }
+
+  private async replaceNestedStacksInParentTemplate(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, generatedParentTemplate: any, deployedParentTemplate: any,
+    parentStackName: string | undefined, listStackResourcesByStackName: { [key: string] : ListStackResources }, sdk: ISDK,
+  ) {
+    const idResourceMap = Object.entries(generatedParentTemplate.Resources ?? {}).filter(([_, attrs]) => this.isNestedStackResource(attrs));
```

Comment:
Sorry, what is `idResourceMap`?

From what I understand, these are nested stacks from the parent template, no? If my understanding is correct, then the name of the variable should reflect that.

Reply:
BTW, if you don't need the keys here, simply use `Object.values()` instead of `Object.entries()`.